packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -43,30 +43,36 @@ As vats run, they send and receive messages. When these messages carry object re
-  `Object`, using the standard objects-as-closures pattern)
+* Heap objects in vat RAM
+  * one-off objects created with `Far()` or `makeExo()`
+  * instances created with a "make" function from `defineExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineExoClassKit()`
 
-* virtual objects (defined with `makeKind()` or `makeDurableKind()`, created
-  by invoking the function those return, and whose state is kept in the
-  database rather than in RAM)
+* Virtual objects in disk-based storage
+  * instances created with a "make" function from `defineVirtualExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineVirtualExoClassKit()`
 
-The v2 code is obligated to re-define all durable virtual-object Kinds created by v1 during the upgrade phase. Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported virtual object.
+* Durable objects in disk-based storage
+  * instances created with a "make" function from `defineDurableExoClass()` or `prepareExoClass()`
+  * multifaceted kit instances created with a "makeKit" function from `defineDurableExoClassKit()` or `prepareExoClassKit()`
+  * singleton objects created with `prepareExo()`
 
-The v2 code is also obligated to reattach any singleton objects. The API for this is still under consideration. The root object may be a special case: if the upgrade invocation is simply a new call to `buildRootObject()`, then whatever object that returns will be associated with the root-object vref (`o+0`).
+During the upgrade phase, v2 code is obligated to re-define all durable Kinds created by v1 (i.e., those associated with objects in the third category). Once complete, this allows liveslots to satisfy deserialization of any inbound message addressed to (or referencing) a previously-exported durable object.
 
-## Durable State
+As a special case, the root object returned from v2's `buildRootObject()` is automatically associated with exportID `o+0` (see [How Liveslots Uses the Vatstore](../../swingset-liveslots/src/vatstore-usage.md#counters)).
 
-The v2 code runs in a brand new JavaScript environment: nothing is carried over from the RAM image of the v1 vat. To fulfill its obligations, v1 must arrange to deliver data and imported object references to v2. This uses two mechanisms: durable storage, and the "baggage" (better name TBD).
+## Durable State
 
-Vat code has access to three categories of collection objects. Each category offers both Map and Set collections, in strong and weak forms (parallel to the standard JavaScript `Map`, `Set`, `WeakMap`, and `WeakSet` facilities).
+The v2 code runs in a brand new JavaScript environment; nothing is carried over from the RAM image of the v1 vat. To fulfill its obligations, v1 must arrange to deliver data and imported object references to v2. This uses two mechanisms: durable storage, and the "baggage" (better name TBD).
```

Comment:
we might as well remove the "better name TBD", I think we're sticking with "baggage"